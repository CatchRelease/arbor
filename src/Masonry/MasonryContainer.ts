import styled from '@emotion/styled';

import Box from '../Box';
import {
  columnCount,
  columnGap,
  columnRule,
  columnRuleColor,
  columnRuleStyle,
  columnRuleWidth,
  columnWidth,
  columns
} from '../utils/columns';

export type Props = {
  columnCount?: number;
  columnGap?: string;
  columnRuleColor?: string;
  columnRuleStyle?: string;
  columnRuleWidth?: string;
  columnWidth?: string;
  columns?: string;
};

const MasonryContainer = styled(Box)<Props>`
  ${columnCount};
  ${columnGap};
  ${columnRule};
  ${columnRuleColor};
  ${columnRuleStyle};
  ${columnRuleWidth};
  ${columnWidth};
  ${columns};
`;

export default MasonryContainer;
