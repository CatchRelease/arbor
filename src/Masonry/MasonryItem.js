import styled from '@emotion/styled';
import { display, width } from 'styled-system';

import Box from '../Box';

const MasonryItem = styled(Box)`
  ${display};
  ${width};
`;

MasonryItem.propTypes = {
  ...display.propTypes,
  ...width.propTypes
};

MasonryItem.defaultProps = {
  display: 'inline-block',
  width: '100%'
};

export default MasonryItem;
