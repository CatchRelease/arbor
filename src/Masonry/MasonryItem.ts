import styled from '@emotion/styled';
import { display, width } from 'styled-system';

import Box from '../Box';

const MasonryItem = styled(Box)`
  break-inside: avoid;
  ${display};
  ${width};
`;

MasonryItem.defaultProps = {
  display: 'block',
  width: '100%'
};

export default MasonryItem;
