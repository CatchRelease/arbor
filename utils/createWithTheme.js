import React from 'react';
import { CacheProvider } from '@emotion/core';
import createCache from '@emotion/cache';
import { create } from 'react-test-renderer';
import { ThemeProvider } from 'emotion-theming';
import theme from '../src/theme';

const myCache = createCache();
myCache.compat = true;

export default (Component) =>
  create(
    <CacheProvider value={myCache}>
      <ThemeProvider theme={theme}>{Component}</ThemeProvider>
    </CacheProvider>
  );
