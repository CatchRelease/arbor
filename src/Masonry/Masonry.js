import PropTypes from 'prop-types';

import MasonryContainer from './MasonryContainer';
import MasonryItem from './MasonryItem';

const Masonry = ({ children, rowGap, ...props }) => (
  <MasonryContainer {...props}>
    {[].concat(children).map((child) => (
      <MasonryItem key={child.key} mb={rowGap}>
        {child}
      </MasonryItem>
    ))}
  </MasonryContainer>
);

Masonry.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node)
  ]).isRequired,
  rowGap: PropTypes.string
};

Masonry.defaultProps = {
  rowGap: undefined
};

export default Masonry;
