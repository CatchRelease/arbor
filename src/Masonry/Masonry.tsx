import { ReactElement } from 'react';
import PropTypes from 'prop-types';

import MasonryContainer, {
  Props as MasonryContainerProps
} from './MasonryContainer';
import MasonryItem from './MasonryItem';

type Props = MasonryContainerProps & {
  children: ReactElement[];
  rowGap?: string;
};

const Masonry = ({ children, rowGap, ...props }: Props) => (
  <MasonryContainer {...props}>
    {children.map((child) => (
      <MasonryItem key={child.key} mb={rowGap}>
        {child}
      </MasonryItem>
    ))}
  </MasonryContainer>
);

Masonry.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node)
  ]).isRequired,
  rowGap: PropTypes.string
};

export default Masonry;
