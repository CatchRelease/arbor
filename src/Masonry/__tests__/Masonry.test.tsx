import Masonry from '../Masonry';
import createWithTheme from '../../../utils/createWithTheme';

describe('<Masonry />', () => {
  it('renders properly', () => {
    const tree = createWithTheme(
      <Masonry
        columnCount={3}
        columnGap="regular"
        columnRuleColor="black"
        columnRuleStyle="solid"
        columnRuleWidth="1px"
        columnWidth="300px"
        rowGap="regular"
      >
        <div key="1">Masonry Item 1</div>
        <div key="2">Masonry Item 2</div>
        <div key="3">Masonry Item 3</div>
        <div key="4">Masonry Item 4</div>
        <div key="5">Masonry Item 5</div>
        <div key="6">Masonry Item 6</div>
        <div key="7">Masonry Item 7</div>
        <div key="8">Masonry Item 8</div>
        <div key="9">Masonry Item 9</div>
      </Masonry>
    );

    expect(tree).toMatchSnapshot();
  });
});
