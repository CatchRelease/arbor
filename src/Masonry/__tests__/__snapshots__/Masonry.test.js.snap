// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Masonry /> renders properly 1`] = `
.emotion-0 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  -webkit-column-count: 3;
  column-count: 3;
  -webkit-column-gap: 1rem;
  column-gap: 1rem;
  -webkit-column-rule-color: #000000;
  column-rule-color: #000000;
  -webkit-column-rule-style: solid;
  column-rule-style: solid;
  -webkit-column-rule-width: 1px;
  column-rule-width: 1px;
  -webkit-column-width: 300px;
  column-width: 300px;
}

.emotion-2 {
  font-family: Motiva Sans;
  margin-bottom: 1rem;
  box-sizing: border-box;
  display: block;
  width: 100%;
  break-inside: avoid;
  display: block;
  width: 100%;
}

<div
  className="emotion-0 emotion-1"
>
  <div
    className="emotion-2 emotion-3"
    display="block"
    width="100%"
  >
    <div>
      Masonry Item 1
    </div>
  </div>
  <div
    className="emotion-2 emotion-3"
    display="block"
    width="100%"
  >
    <div>
      Masonry Item 2
    </div>
  </div>
  <div
    className="emotion-2 emotion-3"
    display="block"
    width="100%"
  >
    <div>
      Masonry Item 3
    </div>
  </div>
  <div
    className="emotion-2 emotion-3"
    display="block"
    width="100%"
  >
    <div>
      Masonry Item 4
    </div>
  </div>
  <div
    className="emotion-2 emotion-3"
    display="block"
    width="100%"
  >
    <div>
      Masonry Item 5
    </div>
  </div>
  <div
    className="emotion-2 emotion-3"
    display="block"
    width="100%"
  >
    <div>
      Masonry Item 6
    </div>
  </div>
  <div
    className="emotion-2 emotion-3"
    display="block"
    width="100%"
  >
    <div>
      Masonry Item 7
    </div>
  </div>
  <div
    className="emotion-2 emotion-3"
    display="block"
    width="100%"
  >
    <div>
      Masonry Item 8
    </div>
  </div>
  <div
    className="emotion-2 emotion-3"
    display="block"
    width="100%"
  >
    <div>
      Masonry Item 9
    </div>
  </div>
</div>
`;
